doc/ipythonconsole.rst

```diff
@@ -2,91 +2,159 @@
+A `Cython console`_ allows you to use the Cython language to speed up your code and call C functions directly from Python.
+Finally, a `Pylab console`_ loads common Numpy and Matplotlib functions by default; while this is deprecated and not recommended for new code, it can be used for older scripts that need it.
+
+.. _Cython console: https://cython.org/#documentation
+.. _Sympy console: https://docs.sympy.org/latest/index.html
+.. _Pylab console: https://matplotlib.org/faq/usage_faq.html#matplotlib-pyplot-and-pylab-how-are-they-related
+
+.. image:: images/console/console-special.gif
+   :alt: Spyder showing opening a new special Console
+
+
+
+============
+Options menu
+============
+
+The options menu allows you to inspect your current environment variables (:guilabel:`Show environment variables`), and the contents of your system's ``PATH`` (:guilabel:`Show sys.path contents`).
+In addition, you can have each console display how long it has been running with :guilabel:`Show elapsed time`.
+
+.. image:: images/console/console-options-menu.png
+   :alt: Spyder IPython Console with options menu
+
+You can also change the name of the current :guilabel:`IPython console` tab with the :guilabel:`Rename tab` option, or by simply double-clicking it.
+
+.. image:: images/console/console-rename.gif
+   :alt: Spyder IPython Console showing renaming console
+
+
+
+.. _connecting-external-kernel:
+
+==================================
+Using external kernels with Spyder
+==================================
+
+You can connect to external local and remote kernels (including those managed by Jupyter Notebook or QtConsole) through the :guilabel:`Connect to an existing kernel` dialog under the :guilabel:`Consoles` menu.
+For this feature to work, a compatible version of the ``spyder-kernels`` package :ref:`must be installed <starting-kernel-problems-ref>` in the environment or machine in which the external kernel is running.
 
 .. image:: images/console/console-menu.png
-   :alt: Spyder IPython Console as above, with the options menu open
+   :alt: Connect to external kernel dialog of the Spyder IPython console
 
 
-Connect to an external kernel
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Connect to a local kernel
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. note::
-   If on Windows and connecting to a remote machine over ``ssh``, you'll need to install the ``paramiko`` python package first, *e.g.* with ``conda install paramiko`` if using Anaconda.
+To connect to a local kernel that is already running (e.g. one started by Jupyter notebook),
 
-To connect to an external kernel,
+#. Run ``%connect_info`` in the notebook or console you want to connect to, and copy the name of its kernel connection file, shown after ``jupyter <app> --existing``.
 
-#. Launch an IPython kernel on the local or remote host if one is not already running.
+   .. image:: images/console/console-connect-local-step1.gif
```

Comment:
I think this gif should start from the point where you can actually read the text shown in the notebook. The notebook interface is easily recognizable by everyone, so I don't see the need to start with showing it in the browser.

That would save like a second and half from this gif.

Reply:
Okay, sure. I can edit it out.

Reply:
Done